### DISCUSSION TITLE
How to BasicAuth for multiple calls / or how to use `btoa`?

### DISCUSSION THREAD

**Discussion by ViToni** [flawiddsouza/Restfox#109] [Q&A] [answered]
I try to use a very simple (local) API which uses Basic Auth. As I didn't find a way to configure Basic Auth (including crendetials) for all calls I tried to use a plugin to do the heavy lifting:

```
function handleRequest() {
    const user = context.request.getEnvironmentVariable('USER');
    const pass = context.request.getEnvironmentVariable('PASSWORD')
    const token = user + ":" + pass;
    const hash = btoa(token);
    const authorization = "Basic " + hash;

    context.request.setRequestHeader("Authorization", authorization);
}
```

Now, when trying to execute any request, I get the error message `ReferenceError: 'btoa' is not defined`.
How can I fix this? Or even better, maybe I'm missing something and I can set up BasicAuth in an easy way.

**Comment by ViToni** [comment]
Just found the function `generateBasicAuthString` in the code base. Using this one would be even better.

**Comment by flawiddsouza** [comment]
Hi, you can use basic auth directly here:
![image](https://github.com/flawiddsouza/Restfox/assets/10164500/7616b028-ed1e-4a5a-8806-21f6dffc4bdb)

**Reply by flawiddsouza** [reply]
Now I see you mentioned "for all calls". I'll see what I can do.

**Reply by ViToni** [reply]
Thanks for caring. Basic auth is just my use case. Have it working would be great (BTW the API in question is [awtrix3](https://github.com/Blueforcer/awtrix3/blob/main/docs/api.md)) and I don't mind having it as a plugin.
I'm wondering though if having something "more generic" lonngterm might help others as well, maybe being able to store the auth type in the environment and then its performed automatically...

**Reply by flawiddsouza** [reply, marked as answer]
You can now set the Auth & Headers for a folder, by right clicking on a Folder -> Properties:

![image](https://github.com/flawiddsouza/Restfox/assets/10164500/3ee9546e-90a2-4828-87ed-23c33b67f030)

**Reply by ViToni** [reply]
@flawiddsouza That's super awesome. Thanks for the swift response and the implementation. Just tested it. Really cool!
